hudi-common/src/main/java/org/apache/hudi/common/config/HoodieFunctionalIndexConfig.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.config;
+
+import org.apache.hudi.common.util.BinaryUtil;
+import org.apache.hudi.common.util.ConfigUtils;
+import org.apache.hudi.exception.HoodieIOException;
+import org.apache.hudi.index.secondary.SecondaryIndexType;
+import org.apache.hudi.metadata.MetadataPartitionType;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import static org.apache.hudi.common.util.ConfigUtils.fetchConfigs;
+import static org.apache.hudi.common.util.ConfigUtils.recoverIfNeeded;
+import static org.apache.hudi.common.util.StringUtils.getUTF8Bytes;
+
+@Immutable
+@ConfigClassProperty(name = "Common Index Configs",
+    groupName = ConfigGroups.Names.INDEXING,
+    subGroupName = ConfigGroups.SubGroupNames.FUNCTIONAL_INDEX,
+    areCommonConfigs = true,
+    description = "")
+public class HoodieFunctionalIndexConfig extends HoodieConfig {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieFunctionalIndexConfig.class);
+
+  public static final String INDEX_DEFINITION_FILE = "index.properties";
+  public static final String INDEX_DEFINITION_FILE_BACKUP = "index.properties.backup";
+  public static final ConfigProperty<String> INDEX_NAME = ConfigProperty
```

Comment:
Are multiple functional indexes supported by the new configs?  e.g., if I'd like to create two indexes based on `month(ts)` with `FILES` and `day(ts)` with `COLUMN_STATS`.

Reply:
yes you can, but you need to run two separate index commands (essentially trigger two different jobs).